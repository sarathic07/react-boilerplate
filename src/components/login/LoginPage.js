import React from "react";
import PropTypes from "prop-types";

class LoginPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      password: ""
    };
    this.inputChange = this.inputChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  inputChange({ target }) {
    this.setState({ [target.name]: target.value });
  }

  onSubmit() {
    this.props.login({
      email: this.state.email,
      password: this.state.password
    });
  }

  render() {
    return (
      <div data-test={"login-comp"}>
        <input
          className={"form-control"}
          name={"email"}
          type="text"
          onChange={this.inputChange}
        />
        <input
          className={"form-control"}
          name={"password"}
          type="password"
          onChange={this.inputChange}
        />
        <input
          className={"form-control"}
          type="button"
          value={"Login"}
          onClick={this.onSubmit}
        />
        <div>{this.props.error}</div>
      </div>
    );
  }
}

LoginPage.propTypes = {
  error: PropTypes.string,
  login: PropTypes.func
};

export default LoginPage;
