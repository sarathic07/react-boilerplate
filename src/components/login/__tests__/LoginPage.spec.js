import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import LoginPage from "../LoginPage";
import { findByTestAtrr } from "./../../../../utils";

const setUp = (props = {}) => {
  const component = shallow(<LoginPage {...props} />);
  return component;
};

describe("Login page", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("Should render without errors", () => {
    const wrapper = findByTestAtrr(component, "login-comp");
    expect(wrapper.length).toBe(1);
  });

  it("match snapshot", () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it("check input elemnts", () => {
    const wrapper = component.find(`input`);
    expect(wrapper.length).toBe(3);
  });
});
