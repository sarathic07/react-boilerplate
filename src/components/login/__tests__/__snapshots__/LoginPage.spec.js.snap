// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page match snapshot 1`] = `
<div
  data-test="login-comp"
>
  <input
    className="form-control"
    name="email"
    onChange={[Function]}
    type="text"
  />
  <input
    className="form-control"
    name="password"
    onChange={[Function]}
    type="password"
  />
  <input
    className="form-control"
    onClick={[Function]}
    type="button"
    value="Login"
  />
  <div />
</div>
`;
